packages/documentation-framework/hooks/useTheme.js

```diff
@@ -0,0 +1,127 @@
+    if (typeof window === 'undefined' || !window.matchMedia) return 'light';
+    
+    if (mode === THEME_MODES.SYSTEM) {
+      try {
+        return window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+      } catch (error) {
+        // Fallback if matchMedia fails
+        console.warn('matchMedia not supported, defaulting to light theme');
+        return 'light';
+      }
+    }
+    return mode;
+  };
+
+  const updateThemeClass = (resolvedTheme) => {
+    if (typeof window === 'undefined' || !document) return;
+    
+    const htmlElement = document.querySelector('html');
+    if (!htmlElement) return;
+    
+    if (resolvedTheme === 'dark') {
+      htmlElement.classList.add(DARK_MODE_CLASS);
+    } else {
+      htmlElement.classList.remove(DARK_MODE_CLASS);
+    }
+  };
+
+  const [themeMode, setThemeModeState] = useState(() => {
+    const stored = getStoredThemeMode();
+    return stored && Object.values(THEME_MODES).includes(stored) ? stored : THEME_MODES.SYSTEM;
+  });
+
+  const [resolvedTheme, setResolvedTheme] = useState(() => getResolvedTheme(themeMode));
+
+  const setThemeMode = useCallback((newMode) => {
+    setThemeModeState(newMode);
+    setStoredThemeMode(newMode);
+    
+    const newResolvedTheme = getResolvedTheme(newMode);
+    setResolvedTheme(newResolvedTheme);
+    updateThemeClass(newResolvedTheme);
+  }, []);
+
+  // Listen for system preference changes
+  useEffect(() => {
+    // Enhanced SSR-safe check
+    if (typeof window === 'undefined' || !window.matchMedia) return;
+
+    let mediaQuery;
+    try {
+      mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+    } catch (error) {
+      console.warn('matchMedia not supported, skipping system theme detection');
+      return;
+    }
+    
+    const handleSystemThemeChange = (e) => {
+      if (themeMode === THEME_MODES.SYSTEM) {
+        const newSystemTheme = e.matches ? 'dark' : 'light';
+        setResolvedTheme(newSystemTheme);
+        updateThemeClass(newSystemTheme);
+      }
+    };
+
+    // Check if addEventListener is available (some older browsers might not support it)
+    if (mediaQuery.addEventListener) {
+      mediaQuery.addEventListener('change', handleSystemThemeChange);
+      return () => {
+        mediaQuery.removeEventListener('change', handleSystemThemeChange);
+      };
+    } else if (mediaQuery.addListener) {
+      // Fallback for older browsers
+      mediaQuery.addListener(handleSystemThemeChange);
+      return () => {
+        mediaQuery.removeListener(handleSystemThemeChange);
+      };
+    }
+  }, [themeMode]);
+
+  // Initial theme application
+  useEffect(() => {
+    const initialResolvedTheme = getResolvedTheme(themeMode);
+    setResolvedTheme(initialResolvedTheme);
+    updateThemeClass(initialResolvedTheme);
+  }, [themeMode]);
+
+  return {
+    themeMode,
+    setThemeMode,
+    resolvedTheme,
+    THEME_MODES
+  };
+};
+
+export { THEME_MODES }; 
```

Comment:
since useTheme is already being exported and returns this, we could avoid exporting this. If we want to export this on its own, though, we should just update line 3 to `export const THEME_MODES ...`